src/Users/Client.php

```
@@ -0,0 +1,109 @@
+use Vonage\Entity\Hydrator\HydratorInterface;
+use Vonage\Entity\IterableAPICollection;
+
+use function is_null;
+
+class Client implements ClientAwareInterface, APIClient
+{
+    use ClientAwareTrait;
+
+    public function __construct(protected APIResource $api, protected ?HydratorInterface $hydrator = null)
+    {
+    }
+
+    public function getApiResource(): APIResource
+    {
+        if (is_null($this->api)) {
+            $api = new APIResource();
+            $api->setClient($this->getClient())
+                ->setBaseUri('/v1/users')
+                ->setCollectionName('users');
+            $this->api = $api;
+        }
+
+        return $this->api;
+    }
+
+    public function getUsers($pageSize = null, $order = null, $cursor = null): IterableAPICollection
+    {
+        if (is_null($pageSize) && is_null($order) && is_null($cursor)) {
+            $filter = new EmptyFilter();
+        }
+
+        if ($pageSize || $order || $cursor) {
+            $filter = new KeyValueFilter([
+                'order' => $order,
+                'cursor' => $cursor
+            ]);
+        }
+
+        $hydrator = new ArrayHydrator();
+        $hydrator->setPrototype(new User());
+
+        $response = $this->api->search($filter);
+        $response->setHydrator($hydrator);
+        $response->setSize($pageSize);
+        $response->setPageSizeKey('page_size');
+        $response->setHasPagination(false);
+
+        return $response;
+    }
+
+    public function createUser(User $user): User
+    {
+        $response = $this->api->create($user->toArray());
+        $userObject = new User();
+        $userObject->fromArray($response);
+
+        return $userObject;
+    }
+
+    public function getUserById(string $id): User
+    {
+        $response = $this->api->get($id);
+        $returnUser = new User();
+        $returnUser->fromArray($response);
+
+        return $returnUser;
+    }
+
+    public function updateUser(User $user, string $id): User
```

Should not need to pass an `$id` for this, because the ID is part of the `User` object. Instead, we should throw an exception if `$user->id` is not set.